drivers/modem/Kconfig.ublox-sara-n310

```diff
@@ -0,0 +1,30 @@
+# u-blox SARA N310 driver options
+
+# Copyright (c) 2019 Foundries.io
+# SPDX-License-Identifier: Apache-2.0
+
+config MODEM_UBLOX_SARA_N310
+	bool "Enable u-blox SARA N310 driver"
+	select MODEM_CONTEXT
+	select MODEM_CMD_HANDLER
+	select MODEM_IFACE_UART
+	imply GPIO
+	help
+	  Choose this setting to enable u-blox SARA-N310
+	  driver.
+
+if MODEM_UBLOX_SARA_N310
+
+config MODEM_UBLOX_SARA_N310_INIT_PRIORITY
+	int "u-blox SARA-N310 driver init priority"
+	default 80
+	help
+	  u-blox SARA-N310 device driver initialization priority.
+	  Do not mess with it unless you know what you are doing.
+	  Note that the priority needs to be lower than the net stack
+	  so that it can start before the networking sub-system.
+	  
+config MODEM_UBLOX_SARA_N310_UART_NAME
```

Comment:
The UART name is used to initiate UART in the driver (i.e. modem_iface_uart_init_dev(&mdata.context.iface, CONFIG_MODEM_UBLOX_SARA_N310_UART_NAME);

This allows the user to specify which UART interface the modem is connected to on an application level (prj.conf). I figured this would be useful alongside letting the user eventually specify things like the APN string for manual selection.